sdk/core/src/main/java/com/klaviyo/coresdk/push/KlaviyoPushService.kt

```diff
@@ -0,0 +1,18 @@
+package com.klaviyo.coresdk.push
+
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.klaviyo.coresdk.utils.ConfigFileUtils
+import com.klaviyo.coresdk.utils.ConfigKeys
+
+class KlaviyoPushService: FirebaseMessagingService() {
```

Comment:
By overriding the FirebaseMessagingService we can listen for changes to the push tokens or for incoming push messages, allowing us to determine what happens on each event. In this case, we need this so we can react to push tokens whenever they are generated (typically if the app is run after a fresh install or if app data was cleared out).

This is Firebase's recommended way of listening for new push tokens.

Reply:
Is this for the app or the sdk? Can you have multiple overrides of this? Customers may use many push services potentially and may need to send to each of them.

Reply:
I'm ok with this in theory but we still might need some API for customers to set tokens. Also we may want to provide a listener here so they can get the to pass to others if necessary.

Reply:
Ya this seems to be the normal way to do things in SDKs with push integration. I did notice that a lot of SDKs have a "manual" mode that disables automatic token generation like this in favour of manual calls to the `FirebaseInstanceId` classes, which will generate tokens async when called.

Reply:
One alternative I was thinking of was to create a `generatePushToken()` call that would check for a stored push token on file, and in the case where none exist, reaching out to `FirebaseInstanceId` to generate and store a fresh one.
We would just want to make sure that we aren't trying to generate push tokens in a getter since the process is async and we're not going to return a token straight away.